lib/shape.ml

```diff
@@ -40,18 +40,24 @@ let rectangle ?x ?y length width =
 
 let ellipse ?x ?y rx ry =
   match (x, y) with
-  | Some x, Some y -> Ellipse {c = {x; y}; rx; ry}
-  | _ -> Ellipse {c = { x = canvas_mid.x; y = canvas_mid.y}; rx; ry}
+  | Some x, Some y -> Ellipse { c = { x; y }; rx; ry }
+  | _ -> Ellipse { c = { x = canvas_mid.x; y = canvas_mid.y }; rx; ry }
+
+let translate dx dy shape =
+  match shape with
+  | Circle circle -> Circle { circle with c = { x = circle.c.x + dx; y = circle.c.y + dy } }
+  | Rectangle rectangle -> Rectangle { rectangle with c = { x = rectangle.c.x + dx; y = rectangle.c.y + dy } }
+  | Ellipse ellipse -> Ellipse { ellipse with c = { x = ellipse.c.x + dx; y = ellipse.c.y + dy } }
 
 let show shapes = List.iter render_shape shapes
 
 let init () =
-  open_graph " 500x500";
+  open_graph " 1000x500";
```

Comment:
Please don't change the canvas dimensions. #19 lets you set the dimensions, if needed.

Reply:
I've merged #19. Could you please rebase your branch on top of main?

Reply:
Okay, thank you. I'll do that now